iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/storable_function.hpp

```diff
@@ -155,7 +154,9 @@ class storable_function<Capacity, signature<ReturnType, Args...>> final
   private:
     operations m_operations; // operations depending on type-erased callable (copy, move, destroy)
 
-    StorageType m_storage;                              // storage for the callable
+    // AXIVION Next Construct AutosarC++19_03-A18.1.1 : safe access is guaranteed since the c-array is wrapped inside the storable_function
```

Comment:
@elBoberido @FerdinandSpitzschnueffler I still think we need the storage abstraction in the future to avoid comments like this etc. I would like to distinguish between allocator and storage as different concepts.

Storage either magically exists or is provided by some other mechanism. In the end it abstracts memory, often but not necessarily contiguous. In a way it wraps memory but abstracts the access and other functionality.

An allocator is a means to get memory, or maybe storage. It can serve multiple allocations and internally generally does not need to manage contiguous memory.

The reason for storage abstraction is also to move to a more general concept where we decouple the storage and the object logic. Anyway, we can keep it for now but I do not see the benefit. Now the function implementation is more tightly coupled again to the storage concept and implements the core functionality (now called `safeAlign`).

Considerations for future refactoring. Resolve.